pkg/controller/flink/flink.go

```diff
@@ -245,10 +245,11 @@ func (f *Controller) StartFlinkJob(ctx context.Context, application *v1alpha1.Fl
 		getURLFromApp(application, hash),
 		jarName,
 		client.SubmitJobRequest{
-			Parallelism:   parallelism,
-			SavepointPath: application.Spec.SavepointInfo.SavepointLocation,
-			EntryClass:    entryClass,
-			ProgramArgs:   programArgs,
+			Parallelism:           parallelism,
+			SavepointPath:         application.Spec.SavepointInfo.SavepointLocation,
+			EntryClass:            entryClass,
+			ProgramArgs:           programArgs,
+			AllowNonRestoredState: application.Spec.AllowNonRestoredState,
```

Comment:
Add to test that this field is populated.
https://github.com/lyft/flinkk8soperator/blob/master/pkg/controller/flink/flink_test.go#L432